src/fparser/two/utils.py

```diff
@@ -1038,62 +1037,84 @@ def tostr(self):
 
 
 class KeywordValueBase(Base):
-    """
-::
-    <keyword-value-base> = [ <lhs> = ] <rhs>
-    """
+    '''
+
+    <keyword-value-base> is [ <lhs> = ] <rhs>
+
+    where:
+
+    R215 keyword is name.
+
+    '''
     @staticmethod
     def match(lhs_cls, rhs_cls, string, require_lhs=True, upper_lhs=False):
         '''
-        :param lhs_cls: list, tuple or single value of classes to attempt to
-                        match LHS against (in order), or string containing
-                        keyword to match
+        Attempts to match the supplied `string` with `lhs_cls` = `rhs_cls`.
```

Comment:
Given that you have improved KeywordValueBase I would expect some changes in the KeywordValueBase tests to either show what is now correct and was not before or perhaps an xfailing test no longer failing. Is there a reason why there are no modifications?

Reply:
That gave me pause for thought but I've found https://github.com/stfc/fparser/issues/253#issuecomment-618248749 which says that the basic problem was that we weren't checking that the LHS of "keyword=value" was actually the keyword that was requested. There is a new test for this.